### PR TITLE
CLI & CMake

### PR DESCRIPTION
**Pull request by revol-xut**
# Things Done
- Cleanup CLI 
- Added Build Backend for CMake
- Lingua-Franca per default is taken from $PATH but can be overwritten with --lfc

# Tested

- [x] Setup project and build directly afterwards 
- [x] Build lf-alarm-clock with it (https://github.com/revol-xut/lf-alarm-clock)

### CODE REVIEW

File: src/backends/cmake.rs [hunk 1]
```diff
@@ -22,6 +22,9 @@ impl Backend for Cmake {
     }
 
     fn build(&self, config: &BuildArgs) -> bool {
+        let create_dir =
+            fs::create_dir_all(format!("{}/build", self.lfc.out.display().to_string())).is_ok();
```
Comment (erlingrj) [inline src/backends/cmake.rs:26]: Is this to build lfc itself? I think it is a bit misleading to have the lfc variable point to the root of the lingua-franca source directory. I think it should either point to the lfc executable or renamed to lf-root or something.

File: src/backends/cmake.rs [hunk 2]
```diff
@@ -22,6 +22,9 @@ impl Backend for Cmake {
...
+
```
Comment (erlingrj) [inline src/backends/cmake.rs:27]: This seems to be Cpp-specific. Either we have to make the backends be language-agnostic, and be extended for specific languages, or we must make them language specific. In which case we would have cmake_c.rs, cmake_cpp.rc. It seems that there are several backends that can be used for both C and Cpp. CMake, GNU Make, Meson, so the first option might be better.

Comment (erlingrj) [reply src/backends/cmake.rs:27]: I am referring here to the build function and in general this CMake backend implementation. Not to this particular diff

File: src/backends/cmake.rs (hunk 1 above)
Comment (revol-xut) [reply src/backends/cmake.rs:26]: This is the directory where the cmake build files go into. To stick with the current folder structure.

File: src/backends/cmake.rs (hunk 2 above)
Comment (revol-xut) [reply src/backends/cmake.rs:27]: Yeah its currently CPP specific. Taking a look how the C cmake works would be one of my next points on my agenda.

File: src/backends/cmake.rs (hunk 1 above)
Comment (erlingrj) [reply src/backends/cmake.rs:26]: Aha, I see.

File: src/backends/cmake.rs (hunk 2 above)
Comment (erlingrj) [reply src/backends/cmake.rs:27]: Agreed, make sense

**Pull request merged**